File: BackEnd/CareerLink/src/main/java/tn/esprit/careerlink/entities/Expense.java
package tn.esprit.careerlink.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults (level = AccessLevel.PRIVATE)
public class Expense implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer idexpense;
    Float unitPrice;
    Float amount;
    Integer quantity;
    private Date dateexpense;
    String category;



    @Enumerated(EnumType.STRING)
    private MethodPayment methodPayment;
    @Enumerated(EnumType.STRING)
    private StatusPayment statusPayment;


    @ManyToOne

    Project project;
    @JsonIgnore
    @OneToOne
    Reclamation reclamation;
    @ManyToOne
    User user;


}

File: FrontEnd/CareerLink1/src/app/models/Expense.ts

import {Project } from "./Project";
import { Reclamation } from "./Reclamation";
import { User } from "./User";

import { MethodPayment } from "./methodPayment";
import { StatusPayment } from "./statuspayment";
export class Expense {

   idexpense!:number;
   unitPrice!: number;
   quantity!: number;
    amount!: number;
    dateexpense!: Date;
    category!: string;
    methodPayment!:MethodPayment;
    
    statusPayment!:StatusPayment;

   user: User = new User();
    project: Project = new Project(); 
    reclamation!:Reclamation[];
    
   
}